nix/nim-status/default.nix

```diff
@@ -0,0 +1,150 @@
+{ newScope
+, mkShell
+, writeScript
+, writeText
+, xcodeWrapper
+, pkgs
+, stdenv
+, lib }:
+let
+  callPackage = newScope {};
+  buildNimStatus = platform: arch: callPackage ./nim-status.nix {
+    inherit platform arch;
+  };
+
+  # Metadata common to all builds of status-go
+  meta = {
+    description = "The Status Go module that consumes go-ethereum.";
+    license = lib.licenses.mpl20;
+    platforms = with lib.platforms; linux ++ darwin;
+  };
+
+  # Source can be changed with a local override from config
+  source = callPackage ./status-go-source.nix { };
+  
+  buildStatusGo = platform: arch: callPackage ./status-go.nix {
+    inherit platform arch meta source;
+  };
+
+  buildAndroid = buildMap: name: stdenv.mkDerivation {
+    name = "${name}-android-builder";
+    buildInputs = [ pkgs.coreutils ];
+    builder = writeScript "${name}-android-builder.sh"
+    ''
+      export PATH=${pkgs.coreutils}/bin:$PATH
+      mkdir $out
+
+      ln -s ${buildMap.x86} $out/x86
+      ln -s ${buildMap.arm} $out/armeabi-v7a
+      ln -s ${buildMap.arm64} $out/arm64-v8a
+    '';
+  };
+
+  # Create a single multi-arch fat binary using lipo
+  # Create a single header for multiple target archs
+  # by utilizing C preprocessor conditionals
+  buildIos = buildMap: name:
+    let
+      headerIos = writeText "${name}.h" ''
+        #if TARGET_CPU_X86_64
+        ${builtins.readFile "${buildMap.x86}/${name}.h"}
+        #elif TARGET_CPU_ARM
+        ${builtins.readFile "${buildMap.arm}/${name}.h"}
+        #else
+        ${builtins.readFile "${buildMap.arm64}/${name}.h"}
+        #endif
+      '';
+    in stdenv.mkDerivation {
+      inherit xcodeWrapper;
+      buildInputs = [ pkgs.coreutils ];
+      name = "${name}-ios-builder";
```

Comment:
1. Why are we inheriting `xcodeWrapper`? `mkDerivation` does not have an argument called `xcodeWrapper`.
2. Why is the `name` not first in the arguments passed to `mkDerivation`?
3. Is the `buildInputs` even necessary if you are putting `coreutils` directly into `PATH` in the script?

Reply:
1. Removed `inherit xcodeWrapper`.
2. Done.
3. Done.